addons/sale_timesheet/views/project_task_views.xml

```diff
@@ -131,13 +131,13 @@
                         column_invisible="not parent.allow_billable"
                         readonly="readonly_timesheet"
                         context="{'with_remaining_hours': True, 'with_price_unit': True}" options="{'no_create': True, 'no_open': True}"
-                        domain="[('is_service', '=', True), ('order_partner_id', 'child_of', parent.partner_id), ('is_expense', '=', False), ('state', '=', 'sale'), ('is_downpayment', '=', False)]"
+                        domain="[('is_service', '=', True), '|', ('order_id.partner_shipping_id.commercial_partner_id.id', 'parent_of', parent.partner_id), ('order_partner_id.commercial_partner_id.id', 'parent_of', parent.partner_id), ('is_expense', '=', False), ('state', '=', 'sale'), ('is_downpayment', '=', False)]"
```

Comment:
this condition `('order_id.partner_shipping_id.commercial_partner_id.id', 'parent_of', parent.partner_id)` should only be added once industry_fsm module is installed.

Reply:
If the `industry_fsm` module is not installed, the `partner_shipping_id` field will still be present in the sale order.

Reply:
It does not mean we manage that use case when industry_fsm module is not installed.

As you can see here: https://github.com/odoo/enterprise/blob/b991f52bf0c57f6a4eec5f3547844876ad80209a/industry_fsm_sale/models/project_task.py#L24

https://github.com/odoo/odoo/blob/68918a457ebf3f5e4a16cf3d9b612d58cf3276d4/addons/sale_project/models/project_task.py#L19

Reply:
The domain in sale_timesheet should be the equivalent than the one defined in the sale_line_id/so_line field to have the same behavior.

Reply:
In the `project.task `model, we have `sale_line_id`, and in the `account.analytic.line` model, we have `so_line`. We need to use `partner_id`, which is a field of the `project.task` model, in the domain for filtering.
So we can't pass the domain from python as we did in `sale_order_id` field.

Reply:
@mjvi-odoo I did not say to remove the domain defined in the xml. I said the domain you wrote is not equivalent to the one defined in the field definition. That is, `('order_id.partner_shipping_id.commercial_partner_id.id', 'parent_of', parent.partner_id)` is not in the domain inside the field definition. That condition is only added into the domain when `industry_fsm` module.